### PR TITLE
Refactor `print_prompt()`

### PR DESCRIPTION
**Pull request by stratact**

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -288,24 +288,13 @@ pub fn set_var(variables: &mut Variables, name: &str, value: &str) {
...
-            print!("+ ");
-        } else {
-            print!("- ");
-        }
-    }
+fn print_prompt(modes: &[Mode]) {
+    let mode = modes.iter().rev().fold(String::new(), |acc, mode| {
+        acc + if mode.value { "+ " } else { "- " }
+    });
+    print!("{}", mode.trim_right());
```
Comment (skylerberg) [inline src/main.rs]: Why are you trimming right?

File: src/main.rs [hunk 2]
```diff
@@ -288,24 +288,13 @@ pub fn set_var(variables: &mut Variables, name: &str, value: &str) {
...
-fn print_prompt(modes: &Vec<Mode>) {
-    for mode in modes.iter().rev() {
-        if mode.value {
-            print!("+ ");
-        } else {
-            print!("- ");
-        }
-    }
+fn print_prompt(modes: &[Mode]) {
+    let mode = modes.iter().rev().fold(String::new(), |acc, mode| {
```
Comment (skylerberg) [inline src/main.rs]: Perhaps naming the string that we print out `mode` is confusing `mode` is also used in the fold. I am thinking something like `mode_prompt` or `prompt_prefix` would be more descriptive.

File: src/main.rs (hunk 1 above)
Comment (stratact) [inline src/main.rs]: because there is an extra space to the right?

File: src/main.rs [hunk 3]
```diff
@@ -288,24 +288,13 @@ pub fn set_var(variables: &mut Variables, name: &str, value: &str) {
...
-    let cwd = match env::current_dir() {
-        Ok(path) => {
-            match path.to_str() {
-                Some(path_str) => path_str.to_string(),
-                None => "?".to_string(),
-            }
-        }
-        Err(_) => "?".to_string(),
-    };
+    let cwd = env::current_dir().ok().map_or("?".to_string(), |ref p| p.to_str().unwrap_or("?").to_string());
```
Comment (skylerberg) [inline src/main.rs]: I like that this gets rid of the duplicate `to_string`s and the duplicate `"?"`. That said, this seems a little like code golf.

Comment (stratact) [inline src/main.rs]: Well, it's pretty much idiomatic and shorter while being concise without the verbose match expressions.

File: src/main.rs (hunk 1 above)
Comment (skylerberg) [inline src/main.rs]: I think that was intentional. If you are in an else branch in an else branch it would say:
`- - user@ion#` or something like that.

Comment (stratact) [inline src/main.rs]: Alright, will fix.

**Pull request merged**